File: solutions/1512. Number of Good Pairs/1512_2.cpp
// Time complexity: O(n)
// Space complexity: O(n)
class Solution {
public:
    int numIdenticalPairs(const vector<int>& nums) {
        unordered_map<int, int> counts;
        int ans = 0;
        
        for (const int num: nums) {
            ans += counts[num];
            counts[num]++;
        }
        
        return ans;
    }
};

File: solutions/1512. Number of Good Pairs/1512_2.js
// Time complexity: O(n)
// Space complexity: O(n)
var numIdenticalPairs = function(nums) {
    const counts = new Map();
    let ans = 0;

    for (const num of nums) {
        ans += counts.get(num) || 0;
        counts.set(num, (counts.get(num) || 0) + 1);
    }

    return ans;
};